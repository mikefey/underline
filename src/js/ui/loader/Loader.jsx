import React from 'react';
import style from './Loader.scss';

/**
 * A loader UI component
 */
const Loader = (props) => {
  const wrapperClassName = props.loaderState.show
    ? style.wrapperShow
    : style.wrapper;

  return (
    <div className={wrapperClassName}>
      <h2 className={style.message}>{props.loaderState.message}</h2>
      <div className={style.loader} />
    </div>
  );
};


/**
 * PropTypes
 * @prop {Object} loaderState - The loader state in the store
 */
Loader.propTypes = {
  loaderState: React.PropTypes.object.isRequired,
};

export default Loader;
