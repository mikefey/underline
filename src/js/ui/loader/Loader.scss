@import '~colors.scss';

.wrapper {
  display: none;
  height: 60px;
  left: 50%;
  margin: -30px 0 0 -30px;
  position: absolute;
  top: 50vh;
  transform: translateZ(0);
  width: 60px;
}

.wrapperShow {
  composes: wrapper;
  display: block;
}

.message {
  float: left;
  font-size: 18px;
  margin: 80px 0 0 -70px;
  position: absolute;
  text-align: center;
  width: 200px;
}

.loader {
  animation: spin .7s infinite linear;
  border-bottom: 4px solid $light-grey;
  border-left: 4px solid $dark-grey;
  border-radius: 50%;
  border-right: 4px solid $light-grey;
  border-top: 4px solid $light-grey;
  height: 100%;
  position: absolute;
  width: 100%;

  &::after {
    border-radius: 50%;
    height: 60px;
    width: 60px;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
