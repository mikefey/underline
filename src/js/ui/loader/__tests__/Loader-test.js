import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import Loader from './../Loader.jsx';

const state = {
  show: true,
  message: '',
};

test('Loader component', (t) => {
  const tree = renderToString(<Loader loaderState={state} />);
  t.snapshot(tree);
});
