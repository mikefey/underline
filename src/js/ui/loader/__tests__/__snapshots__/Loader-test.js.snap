// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component 1`] = `"<div class=\\"Loader__wrapperShow___1BRHEE7C08clRH6 Loader__wrapper___3imVhS9YmKq7eOB\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-593274496\\"><h2 class=\\"Loader__message___37_pDp9j_PMn6_u\\" data-reactid=\\"2\\"></h2><div class=\\"Loader__loader___OksXbA0VgNlfQ7R\\" data-reactid=\\"3\\"></div></div>"`;
